spack test run caliper fails (missing example)

```bash
 spack test run caliper
==> Spack test 2cq47i4gdozyq3hnf4dkiimqnt44vor6
==> Testing package caliper-2.7.0-devjxuj
==> Error: TestFailure: 2 tests failed.


Command exited with status 1:
    '/home/vanessa/Desktop/Code/spack-vsoch/lib/spack/env/gcc/gcc' '/home/vanessa/.spack/test/2cq47i4gdozyq3hnf4dkiimqnt44vor6/caliper-2.7.0-devjxuj/cache/caliper/examples/apps/cxx-example.cpp' '-L/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/caliper-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/lib64' '-I/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/caliper-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/include' '-std=c++11' '-lcaliper' '-lstdc++' '-o' 'cxx-example'
/usr/bin/ld: cannot find -lcaliper
collect2: error: ld returned 1 exit status



1 error found in test log:
     4     ==> [2021-12-13-17:37:29.908894] '/home/vanessa/Desktop/Code/spack-vsoch/lib/spack/env/gcc/gcc' '/home/vanessa/.spack/test/2cq47i4gdozyq3hnf4dk
           iimqnt44vor6/caliper-2.7.0-devjxuj/cache/caliper/examples/apps/cxx-example.cpp' '-L/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubunt
           u20.04-skylake/gcc-9.3.0/caliper-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/lib64' '-I/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu
           20.04-skylake/gcc-9.3.0/caliper-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/include' '-std=c++11' '-lcaliper' '-lstdc++' '-o' 'cxx-example'
     5     /usr/bin/ld: cannot find -lcaliper
     6     collect2: error: ld returned 1 exit status
     7     FAILED: Command exited with status 1:
     8         '/home/vanessa/Desktop/Code/spack-vsoch/lib/spack/env/gcc/gcc' '/home/vanessa/.spack/test/2cq47i4gdozyq3hnf4dkiimqnt44vor6/caliper-2.7.0-de
           vjxuj/cache/caliper/examples/apps/cxx-example.cpp' '-L/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/cali
           per-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/lib64' '-I/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/calip
           er-2.7.0-devjxuj7k3sllecoq74wf2ywszytx2e5/include' '-std=c++11' '-lcaliper' '-lstdc++' '-o' 'cxx-example'
     9     /usr/bin/ld: cannot find -lcaliper
  >> 10    collect2: error: ld returned 1 exit status
     11    
     12      File "/home/vanessa/Desktop/Code/spack-vsoch/bin/spack", line 98, in <module>



Failed to find executable 'cxx-example'

/home/vanessa/Desktop/Code/spack-vsoch/lib/spack/spack/package.py:1882, in run_test:
       1879                    # stack instead of from traceback.
       1880                    # The traceback is truncated here, so we can't use it to
       1881                    # traverse the stack.
  >>   1882                    m = '\n'.join(
       1883                        spack.build_environment.get_package_context(tb)
       1884                    )
       1885


/home/vanessa/Desktop/Code/spack-vsoch/lib/spack/spack/build_environment.py:1051, in _setup_pkg_and_run:
       1048        tb_string = traceback.format_exc()
       1049
       1050        # build up some context from the offending package so we can
  >>   1051        # show that, too.
       1052        package_context = get_package_context(tb)
       1053
       1054        logfile = None

See test log for details:
  /home/vanessa/.spack/test/2cq47i4gdozyq3hnf4dkiimqnt44vor6/caliper-2.7.0-devjxuj-test-out.txt

==> Error: 1 test(s) in the suite failed.

```

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -3,9 +3,10 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import llnl.util.tty as tty
 import os
 import sys
+
+from llnl.util import tty
 
 from spack import *
 

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -5,6 +5,7 @@
 
 import os
 import sys
+import llnl.util.tty as tty
 
 from spack import *
 
@@ -155,7 +156,7 @@ class Caliper(CMakePackage, CudaPackage):
         test_dir = join_path(self.test_suite.current_test_cache_dir, 'examples', 'apps')
 
         if not os.path.isfile(join_path(test_dir, 'cxx-example.cpp')):
-            print('Skipping caliper test: file does not exist')
+            tty.msg('Skipping caliper test: file does not exist')
             return
 
         exe = 'cxx-example'

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -161,14 +161,14 @@ class Caliper(CMakePackage, CudaPackage):
         exe = 'cxx-example'
 
         if os.path.exists(self.prefix.lib):
-            lib_dir = 'lib'
+            lib_dir = self.prefix.lib
         else:
-            lib_dir = 'lib64'
+            lib_dir = self.prefix.lib64
 
         self.run_test(exe='gcc',
                       options=['{0}'.format(join_path(test_dir, 'cxx-example.cpp')),
-                               '-L{0}'.format(join_path(self.prefix, lib_dir)),
-                               '-I{0}'.format(join_path(self.prefix, 'include')),
+                               '-L{0}'.format(lib_dir),
+                               '-I{0}'.format(self.prefix.include),
                                '-std=c++11', '-lcaliper', '-lstdc++', '-o', exe],
                       purpose='test: compile {0} example'.format(exe),
                       work_dir=test_dir)

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -154,15 +154,20 @@ class Caliper(CMakePackage, CudaPackage):
 
         test_dir = join_path(self.test_suite.current_test_cache_dir, 'examples', 'apps')
 
-        if not os.path.exists(test_dir):
-            print('Skipping caliper test')
+        if not os.path.isfile(join_path(test_dir, 'cxx-example.cpp')):
+            print('Skipping caliper test: file does not exist')
             return
 
         exe = 'cxx-example'
 
+        if os.path.exists(self.prefix.lib):
+            lib_dir = 'lib'
+        else:
+            lib_dir = 'lib64'
+
         self.run_test(exe='gcc',
                       options=['{0}'.format(join_path(test_dir, 'cxx-example.cpp')),
-                               '-L{0}'.format(join_path(self.prefix, 'lib64')),
+                               '-L{0}'.format(join_path(self.prefix, lib_dir)),
                                '-I{0}'.format(join_path(self.prefix, 'include')),
                                '-std=c++11', '-lcaliper', '-lstdc++', '-o', exe],
                       purpose='test: compile {0} example'.format(exe),

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -3,9 +3,9 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import llnl.util.tty as tty
 import os
 import sys
-import llnl.util.tty as tty
 
 from spack import *
 

